tdm/docs/architecture.md

```diff
@@ -0,0 +1,222 @@
+   core to consume.
+4. The **Device Attestation** is actually a relying party implementation, for
+   verifying a device attestation evidence and eventually providing attestation
+   results back to the TDM core.
+   
+
+```
+
+                      ┌────────────────────────┐
+                      │ Trusted Device Manager │
+      ┌───────────────┴────────────────────────┴────────────┐
+      │                                     ┌──────────┐    │
+      │                                     │Reference │    │
+      │       ┌───────────┐ ┌───────────┐ ┌─│  Values  │    │
+      │       │  Device   │ │  Device   │ │ └──────────┘    │
+   ┏━━╋━━━━━━▶│ Detection │ │Attestation│◁┤ ┌───────────┐   │
+   ┃  │       │  Module   │ │  Module   │ │ │Attestation│   │
+   ┃  │       └───────────┘ └───────────┘ └─│ Policies  │   │
+   ┃  │             △             △         └───────────┘   │
+   ┃  │             └──────┐┌─────┘                         │
+   ┃  │                    ▽▽                               │
+   ┃  │             ┌──────────────┐                        │
+   ┃  │             │              │                        │
+   ┃  │             │ Core Module  │                        │
+   ┃  │             │              │                        │
+   ┃  │             └──────────────┘                        │
+   ┃  │                     △                               │
+ PCIe │                     └────┐                          │
+ udev │                          │                          │
+events│                          ▽                          │
+   ┃  │         ┌─────────────────────────────────┐         │
+   ┃  │         │       ABI Plugins Module        │         │
+   ┃  │         └─────────────────────────────────┘         │
+   ┃  │         ┌──────────┐┌──────────┐┌─────────┐         │
+   ┃  │         │TDX Plugin││SEV Plugin││SE Plugin│         │
+   ┃  │         └──────────┘└──────────┘└─────────┘         │
+   ┃  │                          ┃                          │
+   ┃  └──────────────────────────╋──────────────────────────┘
+   ┃                             ┃
+   ┃                         CoCo ABI
+   ┃                             ┃
+   ┃                             ▼
+   ┃  ┌─────────────────────────────────────────────────────┐
+   ┗━━│                 Linux Guest Kernel                  │
+      └─────────────────────────────────────────────────────┘
+```
+
+### TDM Core
+
+### Device Detection
+
+### Device Verification
+
+#### TEE ABI Plugins
+
+##### Plugin Interface
+
+A Rust trait called by the TDM core to interact with the TSM.
+(Not needed if Linux provides a converged TEE userspace API.
+
+``` rust
+
+
+
+```
+
+## Theory of Operations
+
+### TEE-IO Acceptance Flow
+
+#### Detection
+
+#### TDI Security Attributes Check
+
+#### Device Attestation
+
+##### Local Attestation
+
+TDM is a Relying Party.
+TDM to use part of the Attestation Service crates
+
+##### Remote Attestation
```

Comment:
For remote attestation, I have a design suggestion: Device attestation module of TDM can be responsible for call `get_token` API of Attestation-Agent (We are about to implement this API) and pass the device evidence to AA to perform remote attestation. Then TDM only need to verify the signature of the attestation result token. This aligns with AA's now role in guest components.

Reply:
@jialez0 Would the TDM provide the attestation agent with the device attestation evidence? And then let the AA run remote attestation to finally get an attestation results token? If so, that makes sense to me, and I started to add something like that in this document.

Reply:
@sameo Now AA has two interfaces: Get Evidence and Get Token
I think a better design would be:
- In local attestation mode: TDM Attestation Module is responsible for calling AA's Get Evidence API to obtain Evidence and verifying evidence with locally integrated AS.
- In remote attestation mode: TDM Attestation Module is responsible for calling AA's Get Token API to perform remote attestation and ultimately obtain an attestation results token.

In both modes, the collection of Attestation Evidence is completed by the integrated attester drivers in AA. So that, to support device attestation we only need to add a new attester for AA and a new verifier for AS. Just like we support a new type of TEE.